### ISSUE TITLE
Session out again

### ISSUE THREAD

**Issue by baek-sang** [bitwarden/clients#10718]
### Steps To Reproduce

You have a fresh installation of the Firefox extension.

Logout occurred while the browser was still working (2 different PC)

### Expected Result

Keeping sessions alive

### Actual Result

logout

### Additional Context

https://github.com/bitwarden/clients/issues/8873

### Operating System

Windows

### Operating System Version

11

### Web Browser

Firefox

### Browser Version

130

### Build Version

2024.7.1

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by trmartin4** [comment]
@baek-sang Thank you for reporting this issue.  To help us reproduce and diagnose the problem, could you please provide some additional information?

- Are you logged in to multiple accounts on the extension when this occurs?
- What are your vault timeout settings?
  - Vault timeout length
  - Vault timeout action
- Does this seem to occur only after a certain period of time (e.g. 1 hour of no activity)?

Thank you for your help.

**Comment by trmartin4** [comment]
@baek-sang hello!  Have you continued to experience this logout at any interval over the past few months?  We have not had reports of this elsewhere and want to see if it has potentially been resolved.

**Comment by trmartin4** [comment]
@baek-sang Thank you for your reply.  Could you clarify what you mean by "updating a self-hosted server"?  What is being done when you experience a logout?  Are you saying that you are logged in to a self-hosted instance on your Firefox extension, update your self-hosted server version, and are logged out?

**Comment by trmartin4** [comment]
@baek-sang thank you!  So, to summarize, you are experiencing logout after updating the version of your Unified self-hosted instance?

**Comment by trmartin4** [comment]
Hello.  Can you confirm whether you experience the same behavior with an upgrade to `2024.11.0`, which was released this week?

**Comment by dgrinin** [comment]
I have the same issue with the Safari extension. However, I must admit that this issue only occurs on one of my two Macs (MacBook Pro with Intel and Mac Mini M2). I only experience this problem with the Mac Mini. One possible reason I can think of is that I have only one user on the MacBook, but multiple users on the Mac Mini.

**Comment by AndrewLauu** [comment]
I got the same question when i was in 2024.11.5 (19461 `beta version`), then I reinstall the 2024.11.6 (19499 `standard version`) and it was ok. #4318
Now I upgraded to 2024.11.6 (19499 `com.x8bit.bitwarden-standard-release.apk `) it went bad and log me out again.

**Comment by SergeantConfused** [comment]
Hello @dgrinin and @AndrewLauu,

Are you using a self-hosted Bitwarden server, or the Bitwarden cloud infrastructure (vault.bitwarden.com/eu)?

Thank you in advance,

**Comment by MineGene** [comment]
Also experiencing this, support ticket: #574006.

**Comment by SergeantConfused** [comment]
Hello everyone,

Is anyone still encountering this with Bitwarden browser extension 2024.12.4 or newer?


Thank you,

**Comment by MineGene** [comment]
I will next be able to check 24th-27th Jan, I’ll update my findings then.

**Comment by MineGene** [comment]
@SergeantConfused Issue still exists for me.

**Comment by markrosan8** [comment]
This has been an issue since Aug 2024, and still unfixed?

**Comment by baek-sang** [comment]
Is there any progress on this bug? The logout issue also occurred in Firefox 2025.6.1. I've been logged out multiple times, and I believe it happened when I upgraded my self-hosted version using `docker compose pull && docker compose up -d`.

Sometimes, I'm logged out for no apparent reason, which seems to occur when the server is offline. This is very inconvenient because the logout issue happens not only with the Firefox addon but also with the Windows portable app.

**Comment by hmoffatt** [comment]
I am experiencing this with the current versions of Firefox (144.0) and Firefox extension (2025.10), on Windows. I have not seen it on my other computers though. I have uninstalled the extension and reinstalled.

Overnight the browser extension gets logged out. Timeout action is to lock, Timeout is on browser restart.

The log from when it was working last night, and then logged out this morning.

It looks like the computer (Windows 11) woke up by itself at 09:05 this morning, I didn't start using it until 09:29. Firefox was open the whole time.


```
21:04:25.038 [SignalR] Close message received from server. console-log.service.ts:43:17
21:04:25.040 [SignalR] Connection disconnected with error 'Error: Server returned an error on close: Connection closed with an error.'. console-log.service.ts:51:17

09:05:17.655 XHRGET https://api.bitwarden.com/accounts/revision-date [HTTP/2 401  1430649ms]
09:29:07.552 Firefox can’t establish a connection to the server at wss://notifications.bitwarden.com/hub?access_token=... vendor.js:2:37879

09:29:07.560 [SignalR] (WebSockets transport) There was an error with the transport. console-log.service.ts:43:17
09:29:07.561[SignalR] Failed to start the connection: Error: WebSocket failed to connect. The connection could not be found on the server, either the endpoint may not be a SignalR endpoint, the connection ID is not present on the server, or there is a proxy blocking WebSockets. If you have multiple servers check that sticky sessions are enabled. console-log.service.ts:51:17

09:29:09.084 [PhishingDetectionService] No active account. console-log.service.ts:43:17
09:29:09.098 Retrieving application id

09:29:09.902 Environment did not respond in time, emitting previous config. console-log.service.ts:43:17
09:29:09.902 [PhishingDetectionService] No active account. console-log.service.ts:43:17
09:29:10.086 can't access property "id", e is null console-log.service.ts:47:17

09:29:10.949 GET wss://notifications.bitwarden.com/hub?access_token=...
[HTTP/1.1 401 Unauthorized 3544ms]

09:29:11.052 can't access property "id", e is null console-log.service.ts:47:17
09:29:11.187 Cannot find menu item with id autofill console-log.service.ts:47:17
09:29:11.374 can't access property "id", e is null console-log.service.ts:47:17
09:29:11.451 Cannot find menu item with id copy-password console-log.service.ts:47:17
09:29:12.098 [PhishingDetectionService] Initialize called. Checking prerequisites... console-log.service.ts:43:17
09:29:12.112 WebAssembly is supported in this environment browser-sdk-load.service.ts:51:13
09:29:12.120 Retrieving application id console-log.service.ts:43:17
09:29:12.182 Uncaught (in promise) Error: Could not establish connection. Receiving end does not exist.
    Fx index.ts:116
background.js:1:1035362

09:29:12.279 WASM SDK loaded in 166ms console-log.service.ts:43:17
09:29:12.284 State version: 72
```

**Comment by trmartin4** [comment]
@hmoffatt is this something that occurs regularly?

The issue appears to be this line in the logs:
```
09:05:17.655 XHRGET https://api.bitwarden.com/accounts/revision-date [HTTP/2 401  1430649ms]
```

This request took 1430649ms (almost 24 minutes), and what most likely happened was that the access token used to make the request was no longer valid when it actually executed, which the application responded to by logging you out.

Can you provide more detail on exactly what mode the Windows 11 machine was in overnight?  You said that it "woke up by itself" - which of the [modes](https://support.microsoft.com/en-us/windows/shut-down-sleep-or-hibernate-your-pc-2941d165-7d0a-a5e8-c5ad-8c972e8e6eff) was it in overnight?

What I believe happened was that:
1. At 9:05 AM, a background sync was attempted.  This initiated a request to the Bitwarden server with the valid access token _as of 9:05 AM_.  Bitwarden's access tokens are short-lived (1 hour).
2. Due to the state of the machine at the time, the request didn't actually execute until you logged in at 9:26 AM.
3. By 9:26 AM, that access token was no longer valid.

While we investigate further, any information you can provide about the state of the machine overnight might help us reproduce and fix this.

Thank you!

**Comment by baek-sang** [comment]
- Firefox 145
- Extension 2025.10.0

The logging-out error is still occurring.
The problem is occurring even though I haven't upgraded the server and I am not offline.

Furthermore, I had never been logged out of the mobile app before, but I was logged out after the iOS 26 upgrade. The Windows Portable App used to log me out every time the version was upgraded; although the current version (2025.11.0) doesn't have this issue yet, it is a very frustrating problem for this error to occur intermittently.

**Comment by hmoffatt** [comment]
> [@hmoffatt](https://github.com/hmoffatt) is this something that occurs regularly?

On one particular computer, it happens every day. I generally login again in the morning, the computer sleeps overnight, and I'm logged out again next morning.


> What I believe happened was that:
> 
>     1. At 9:05 AM, a background sync was attempted.  This initiated a request to the Bitwarden server with the valid access token _as of 9:05 AM_.  Bitwarden's access tokens are short-lived (1 hour).
> 
>     2. Due to the state of the machine at the time, the request didn't actually execute until you logged in at 9:26 AM.
> 
>     3. By 9:26 AM, that access token was no longer valid.

I agree, from my reading of the console. I don't know why the computer woke at 9:05 AM, it should still have been asleep, though Windows can be odd in this respect.

This doesn't happen on my other Windows PC, which also sleeps overnight, nor my Linux machines. I am away from that computer for a while and unfortunately not able to get any more information from it for now.

**Comment by hmoffatt** [comment]
Great timing, I will be able to test this on my troublesome desktop in a couple of weeks.

**Comment by hmoffatt** [comment]
This seems to be still happening for me in 2025.12.1.

**Comment by MineGene** [comment]
It happened last night for me also.

**Comment by hmoffatt** [comment]
Unfortunately I can confirm it is still definitely happening for me with 2025.12.1. @trmartin4 what debug can I provide?

**Comment by hmoffatt** [comment]
I've opened a new issue about this, since it persists. I've also noticed that the extension is now unresponsive when I login again, until I disable and re-enable it, or restart Firefox.

**Comment by pamperer562580892423** [comment]
@hmoffatt @MineGene As far as I can see, the [fix](https://github.com/bitwarden/clients/pull/17407) just wasn't included in the last release.

PS:

https://github.com/bitwarden/clients/pull/17407 (the fix) was merged on Jan 6:

<img width="922" height="269" alt="Image" src="https://github.com/user-attachments/assets/c323e33f-0ba0-4e8d-ad15-5947286dcf67" />

&nbsp;

You can see that it wasn't included, when comparing browser extension 2025.12.1 (latest version) with 2025.12.0: https://github.com/bitwarden/clients/compare/browser-v2025.12.0...browser-v2025.12.1

... as that's the end of the list - and the fix is not listed:

<img width="1540" height="615" alt="Image" src="https://github.com/user-attachments/assets/d5912c33-5355-4161-976c-2e47996132bd" />

**Comment by hmoffatt** [comment]
Apologies, I did not realize that this wasn't released yet. I tried installing a new build (from the github actions artifacts) but Firefox refused to install an unverified addon.

**Comment by baek-sang** [comment]
- Bitwarden-Portable-2026.1.0
logout

```
[2026-02-13 14:29:23.683] [info]  Using SignalR for server notifications
[2026-02-13 14:29:23.687] [info]  Retrieving application id
[2026-02-13 14:29:23.727] [error] Error refreshing access token:  { response: { error: 'invalid_grant' }, statusCode: 400 }
[2026-02-13 14:29:23.727] [error] [SignalR] Failed to start the connection: [object Object]
[2026-02-13 14:29:25.757] [info]  [Pin Service] Pin-unlock via PinProtectedUserKeyEnvelope
[2026-02-13 14:29:25.978] [info]  [PinService]: UnsealPinEnvelope took 220.80000001192093 undefined
[2026-02-13 14:29:25.984] [info]  Vault unlocked 2026-02-13T05:29:25.984Z
[2026-02-13 14:29:26.043] [error] Unable to fetch ServerConfig from https://mydomain/api { response: { error: 'invalid_grant' }, statusCode: 400 }
[2026-02-13 14:29:26.127] [info]  [LockComponent] Sync took 35ms
[2026-02-13 14:29:26.127] [info]  [LockComponent] Private key regeneration took 0ms
```

**Comment by pamperer562580892423** [comment]
@baek-sang 

> Unable to fetch ServerConfig from https://mydomain/api

Are you self-hosting your Bitwarden server?

PS: Ah, I now saw further above that you are self-hosting. Then, what is your current server version? - And that may also be a configurational issue then.

**Comment by baek-sang** [comment]
@pamperer562580892423 

mydomain(masking real domain)
self-host - ghcr.io/bitwarden/lite:2026.1.1

The Firefox extension (v2025.12.1) appears to have logged out of two profiles at the same time. However, the iOS app remains logged in. My self-hosted server has not experienced any downtime.

**Comment by pamperer562580892423** [comment]
@baek-sang 

> The Firefox extension (v2025.12.1) appears to have logged out of two profiles at the same time.

FWIW, the proposed fix here was not included in extension 2025.12.1. (see [here](https://github.com/bitwarden/clients/issues/10718#issuecomment-3802710673))

**Comment by baek-sang** [comment]
@pamperer562580892423 

I know.

But Windows Desktop client version is *v2026.1.0*

Hasn't the patch been applied to the desktop version yet?

**Comment by pamperer562580892423** [comment]
@baek-sang:

> But Windows Desktop client version is _v2026.1.0_
> 
> Hasn't the patch been applied to the desktop version yet?

The fix is listed for 2026.1.0... but probably only someone from Bitwarden could answer if it is "live" or not. 🤷‍♂️
